x/ethbridge/types/claim.go

```diff
@@ -72,28 +64,39 @@ func NewOracleClaimContent(
 // must be created in a deterministic way that all validators can follow.
 // For this, we use the Nonce an Ethereum Sender provided,
 // as all validators will see this same data from the smart contract.
-func CreateOracleClaimFromEthClaim(cdc *codec.Codec, ethClaim EthBridgeClaim) (oracle.Claim, error) {
-	oracleID := strconv.Itoa(ethClaim.EthereumChainID) + strconv.Itoa(ethClaim.Nonce) + ethClaim.EthereumSender.String()
-	claimContent := NewOracleClaimContent(ethClaim.CosmosReceiver, ethClaim.Amount,
-		ethClaim.Symbol, ethClaim.TokenContractAddress, ethClaim.ClaimType)
+func CreateOracleClaimFromEthClaim(ethClaim *EthBridgeClaim) (oracletypes.Claim, error) {
+	oracleID := strconv.FormatInt(ethClaim.EthereumChainId, 10) + strconv.FormatInt(ethClaim.Nonce, 10) +
+		ethClaim.EthereumSender
+
+	cosmosReceiver, err := sdk.AccAddressFromBech32(ethClaim.CosmosReceiver)
+	if err != nil {
+		return oracletypes.Claim{}, err
+	}
+
+	claimContent := NewOracleClaimContent(cosmosReceiver, ethClaim.Amount,
+		ethClaim.Symbol, NewEthereumAddress(ethClaim.TokenContractAddress), ethClaim.ClaimType)
 	claimBytes, err := json.Marshal(claimContent)
 	if err != nil {
-		return oracle.Claim{}, err
+		return oracletypes.Claim{}, err
 	}
 	claimString := string(claimBytes)
-	claim := oracle.NewClaim(oracleID, ethClaim.ValidatorAddress, claimString)
+	claim := oracletypes.NewClaim(oracleID, ethClaim.ValidatorAddress, claimString)
 	return claim, nil
 }
 
 // CreateEthClaimFromOracleString converts a string
 // from any generic claim from the oracle module into an ethereum bridge specific claim.
 func CreateEthClaimFromOracleString(
-	ethereumChainID int, bridgeContract EthereumAddress, nonce int,
-	ethereumAddress EthereumAddress, validator sdk.ValAddress, oracleClaimString string,
-) (EthBridgeClaim, error) {
+	ethereumChainID int64,
+	bridgeContract EthereumAddress,
+	nonce int64,
+	ethereumAddress EthereumAddress,
+	validator sdk.ValAddress,
+	oracleClaimString string,
+) (*EthBridgeClaim, error) {
```

Comment:
Same question here: why do we use a pointer to an EthBridgeClaim instead of just a regular EthBridgeClaim.